app/Http/Controllers/Transfer/User/SendController.php

```diff
@@ -2,21 +2,107 @@
 
 namespace App\Http\Controllers\Transfer\User;
 
+use App\Classes\WalletManager;
+use App\Transformers\TransferTransformer;
+use App\Wallet;
+use App\Transfer;
+use Money\Money;
+use Money\Currency;
 use Illuminate\Http\Request;
 use App\Http\Controllers\Controller;
+use App\Transformers\WalletTransformer;
+use App\Http\Responses\JsonErrorResponse;
 
 class SendController extends Controller
 {
 
     /**
      * Creates a sending transfer
      *
-     * @param  \Illuminate\Http\Request  $request
-     * @return \Illuminate\Http\Response
+     * @param Request $request
+     * @return JsonErrorResponse|array
      */
     public function send(Request $request)
     {
-        //
+        $this->validateRequest($request);
+
+        $user = $request->user();
+
+        $wallet = Wallet::find($request->wallet_id);
+        if (! $wallet || $wallet->user_id != $user->id) {
+            return $this->sendErrorResponse('Wallet does not exists.');
+        }
+
+        $integer = $wallet->currency->toInteger($request->amount);
+        if (((int) $integer) - $integer < 0){
```

Comment:
I think laurendy's way is right. The alternative is:  strlen(substr(strrchr($str, "."), 2)); which is more complicated.